README.md

```diff
@@ -1,9 +1,126 @@
+# kotsu: lightweight framework for structuring model validation
+
 [![PyPI version](https://img.shields.io/pypi/v/kotsu.svg)](https://pypi.org/project/kotsu/)
 ![lint-test status](https://github.com/datavaluepeople/kotsu/actions/workflows/run-ci.yml/badge.svg?branch=main)
 [![codecov](https://codecov.io/gh/datavaluepeople/kotsu/branch/main/graph/badge.svg?token=3W8T5OSRZZ)](https://codecov.io/gh/datavaluepeople/kotsu)
 
-# kotsu
-Lightweight framework for structured and repeatable model validation
+## What is it?
+
+**kotsu** is Python package that provides a lightweight and flexible framework to structure
+validating and comparing machine learning models. It aims to provide the skeleton on which to
+develop any kinds of models and to validate them in a robust and repeatable way, **without bloat or
+overheads**. Its flexibility fully leaves open usage with **any model interface**, and any
+complexity of validation, while the structure it provides **avoids common pitfalls** that occur
+when attempting to make fair comparisons between models.
+
+## Main Feature
+
+  - Register a model with hyperparameters to a unique ID
+  - Register validations to a unique ID
+  - Run all registered models through all registered validations, and have the results compiled and
+    stored into a dataframe
+  - Optionally passes an `artefacts_directory` to your validations, for storing of outputs for
+    further analysis, e.g. trained models, model predictions on test data sets
+  - Doesn't enforce any constraints or requirements on your models' interfaces
+  - Pure Python package, with no other setup or configuration of other systems required
+
+## Usage
+
+Import your packages for modelling and kotsu:
+
+```python
+from sklearn import datasets, svm
+from sklearn.model_selection import cross_val_score
+
+import kotsu
+```
+
+Register your competing models. Here we register two Support Vector Classifiers with different
+hyper-parameters:
+
+```python
+model_registry = kotsu.registration.ModelRegistry()
+
+model_registry.register(
+    id="SCV-v1",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 1, "random_state": 1},
+)
+
+model_registry.register(
+    id="SCV-v2",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 0.5, "random_state": 1},
+)
+```
+
+Register your validations. You can register multiple if you want to compare models over different
+tests, e.g. on different datasets. Your validation should take an instance of your models as an
+argument and return a dictionary containing the results. Here we register two Cross-Validation
+validations with different numbers of folds:
+
+```python
+validation_registry = kotsu.registration.ValidationRegistry()
+
+
+def factory_iris_cross_valdation(folds: int):
+    """Factory for iris cross validation."""
```

Comment:
a general point about factories: it suddenly strikes me that the validations might tend to always just be functions (ie, not have any state), in which case the extra hoop being jumped through about making factories (rather than just writing the function) might be annoying - but i suggest we come back to this after using it a little to see how it feels

Reply:
wondering if this is actually almost an example of where there is state in the sense of the data needs to be loaded and that is currently being done within the validation function body but could be done in the constructor and potentially should be done in the constructor e.g. if we want to share the folds between different models and they are randomly created. As a middle ground validations could be callable class instances?

Reply:
> it suddenly strikes me that the validations might tend to always just be functions

Yea might well turn out to be the case. We can see how it looks once we're using it in the wild ay, and update the docs with whatevers more indicative once we know. 

Added this to #19 

> wondering if this is actually almost an example of where there is state in the sense of the data needs to be loaded and that is currently being done within the validation function body but could be done in the constructor

Yea could well be. I'm super certain how it is going to look in the wild yet. But certainly will be something I'll be acutely thinking about during usage of it. 

> As a middle ground validations could be callable class instances or just functions?

I'm pretty sure that they can already be either of those with this currently implementation.